doc/contributing/hw/comportability/README.md

```diff
@@ -462,62 +462,137 @@ Here is an example specification:
-All interrupts are sent to the processor as active-high level (as opposed to edge) interrupts.
-Events themselves can be edge or level, active high or low, as defined by the associated peripheral.
-For instance, the GPIO module might detect the rising or falling edge of one its input bits as an interrupt event.
+Currently, all wired CIP interrupts in OpenTitan use active-high level-triggering.
+(An alternate standard is the use of edge-triggered interupts, which are not described here, but may be supported in a future release.)
 
-The latching of the event is done by the auto-generated register file as described below.
-The clearing of the event is done by a processor write when the handling of the event is completed.
-The waveform below shows the timing of the event occurrence, its latched value, and the clearing by the processor.
-More details follow.
+![OpenTitan System Interrupt Architecture](ot_interrupt_arch.svg)
+
+Interrupts sent to the processor are aggregated by a platfrom level interrupt controller (PLIC).
+Within that logic there may be another level of control for enabling, prioritizing, and enumeration.
+Specification of this control is defined in the rv_plic documentation of the corresponding toplevel design.
+
+### Defining Interrupts
+
+The `.hjson` [configuration file](#configuration-file) for a peripheral specifies interrupts in the field 'interrupt_list'.
+See the [example](#configuration-information-in-the-file) file above for one possible use of this schema.
+For each entry in 'interrupt_list', the following keys are defined:
+
+Key | Kind | Type | Description of Value
+--- | ---- | ---- | --------------------
+name | required | string | Name of the interrupt
+desc | optional | string | Description of the interrupt
+type | optional | string | Select between "Event" and "Status" types, default is "Event" if unspecified.
+
+### Interrupts per module
+
+A peripheral generates a separate wired interrupt for each event and sends them all as bundle to the target's interrupt controller.
+All peripheral interrupts have equal severity, but may be classified into a heirarchy at a higher level, such as at a platform interrupt controller.
+The determination of which input has interrupted the processor, sometimes referred to as "Disambiguation", is done primarily by querying the rv_plic claim/complete register (`CC`).
+This is distinct from another possible model in which each peripheral would send only one interrupt, and the processor would disambiguate by querying the peripheral to figure out which interrupt was triggered.
+
+### Register Creation
+
+By default for every peripheral, three registers are **automatically** created to manage the interrupts (as defined in the `interrupt_list` portion of the Hjson file).
+These registers are named `INTR_STATE`, `INTR_ENABLE` and `INTR_TEST`, and are placed at the top of the peripheral's address map in that order automatically by the `reggen` tool.
```

Comment:
Grammar nit: It's probably more comfortable to read if this is "... and are automatically placed ..."